compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -1378,6 +1408,82 @@ int32_t TR_OSRExceptionEdgeRemoval::perform()
          }
       }
 
+   // Under voluntary OSR, it is common for all paths to an OSR code block to contain
+   // a dead store for some of the prepareForOSR loads. As this is a cold path and
+   // OSR infrastructure is commonly ignored, the optimizer won't eliminate these loads.
+   //
+   // As this is cheap to do and will reduce the overhead of analysis like UseDef and
+   // CompactLocals, identify the loads that have dead stores on all paths
+   // and zero them.
+   //
+   static const char *disableDeadStoreCleanup = feGetEnv("TR_DisableDeadStoreCleanup");
+   if (!disableDeadStoreCleanup)
+      {
+      TR_BitVector alwaysDead(comp()->trMemory()->currentStackRegion());
+      TR_Stack<TR::Block *> osrBlocks(trMemory(), 8, false, stackAlloc);
```

Comment:
A comment on why we can't process when we find them with a bitvector of already processed blocks would be handy here if this is the case...

Reply:
Added the comment. The blocks are revisited to clean up dead stores. A stack isn't required, but it seems like a cheap solution.